R/unnecessary_lambda_linter.R

```diff
@@ -131,6 +186,27 @@ unnecessary_lambda_linter <- function() {
       type = "warning"
     )
 
+    inner_comparison_lints <- NULL
+    if (!allow_comparison) {
+      inner_comparison_expr <- xml_find_all(xml, inner_comparison_xpath)
+
+      mapper <- xp_call_name(xml_find_first(inner_comparison_expr, "parent::expr/parent::expr"))
+      if (length(mapper) > 0L) fun_value <- if (mapper == "sapply") "" else ", FUN.VALUE = <intermediate>"
+
+      inner_comparison_lints <- xml_nodes_to_lints(
+        inner_comparison_expr,
+        source_expression = source_expression,
+        lint_message = sprintf(
+          paste(
+            "Compare to a constant after calling %1$s() to get the full benefits of vectorization.",
+            "Prefer %1$s(x, foo%2$s) == 2 over %1$s(x, function(xi) foo(xi) == 2, logical(1L))."
```

Comment:
Can we find the actual `foo`, `==` and `2` from the lint?

Reply:
OK to push to follow-up?

Reply:
If it's a lot of work, yes.

Reply:
Enough, and we're deep enough into initial work on this feature, that I think it's better to bump: #2323